modules/workergroup/instanceconfig.tf

```diff
@@ -0,0 +1,100 @@
+# Copyright (c) 2022, 2023 Oracle Corporation and/or its affiliates.
+# Licensed under the Universal Permissive License v 1.0 as shown at https://oss.oracle.com/licenses/upl
+
+resource "oci_core_instance_configuration" "instance_configuration" {
+  # Create an OCI Instance Configuration resource for each enabled entry of the worker_groups map with a mode that uses one.
+  for_each       = local.enabled_instance_configs
+  compartment_id = each.value.compartment_id
+  display_name   = "${each.value.label_prefix}-${each.key}"
+
+  instance_details {
+    instance_type = "compute"
+
+    launch_details {
+      # Define each configured availability domain for placement, with bounds on # available
+      # Configured AD numbers e.g. [1,2,3] are converted into tenancy/compartment-specific names
+      availability_domain = lookup(local.ad_number_to_name, (
+        contains(keys(each.value), "placement_ads")
+        ? element(tolist(setintersection(each.value.placement_ads, local.ad_numbers)), 1)
+        : element(local.ad_numbers, 1)
+      ), "")
+      compartment_id = each.value.compartment_id
+      defined_tags = merge(
+        local.defined_tags,
+        lookup(each.value, "defined_tags", {}),
+      )
+      freeform_tags = merge(local.freeform_tags, contains(keys(each.value), "freeform_tags") ? each.value.freeform_tags : { worker_group = each.key })
+
+      instance_options {
+        are_legacy_imds_endpoints_disabled = false
+      }
+
+      create_vnic_details {
```

Comment:
In case NPN is selected, multiple VNICs may need to be created. How is this handled here? I see only 1 vnic

Reply:
Secondary VNICs for NPN are attached by CCM

Reply:
Added preconditions for supported `cni_type`